swc/defs.bzl

```diff
@@ -112,3 +116,30 @@ def swc(name, srcs = None, args = [], data = [], output_dir = False, swcrc = Non
         root_dir = root_dir,
         **kwargs
     )
+
+_swc_plugin = rule(
+    doc = "Configure an SWC plugin",
+    implementation = _swc_plugin_lib.implementation,
+    attrs = _swc_plugin_lib.attrs,
+    provides = _swc_plugin_lib.provides,
+)
+
+def swc_plugin(name, src = None, config = None, **kwargs):
+    """Configure an SWC plugin
+
+    Args:
+        name: A name for this target
+
+        src: Label for the plugin, either a directory containing a package.json pointing at a wasm file
+            as the main entrypoint, or a wasm file.
+
+        config: Configuration dict for the plugin.
+
+        **kwargs: additional keyword arguments passed through to underlying rule, eg. `visibility`, `tags`
+    """
+    _swc_plugin(
+        name = name,
+        src = src,
+        config = json.encode(config),
```

Comment:
needs a check that `config` is a dict, use sth like https://github.com/bazelbuild/bazel-skylib/blob/main/docs/types_doc.md#typesis_dict